lib/connectors/memory.js

```diff
@@ -723,6 +725,61 @@ Memory.prototype.updateAttributes = function updateAttributes(model, id, data, o
   }
 };
 
+Memory.prototype.replaceById = function(model, id, data, options, cb) {
+  var self = this;
+  if (!id) {
+    var err = new Error('You must provide an id when replacing!');
+    return cb(err);
+  }
+
+  // Do not modify the data object passed in arguments
+  // data = Object.create(data);
```

Comment:
> Take a look at L712/714 - it uses the same trick. I don't see why this should not work here.

Still I'm not sure why this line needs to be commented out; I'm using exactlty the same trick as [here](https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/connectors/memory.js#L712-L714)